File: packages/dd-trace/test/llmobs/parity/scenarios/langchain/chat-openai-error.js
'use strict'

const { finish } = require('../common')
const helpers = require('./helpers')

async function main () {
  const tracer = helpers.loadLangChainTracer()
  const model = helpers.chatOpenAI({ temperature: 0, maxRetries: 0 })
  try {
    await model.invoke([['user', 'This request fails.']])
  } catch {
    // expected: the stub returns a 400
  }
  await finish(tracer)
}

main().catch(error => {
  process.stderr.write(`${error.stack}\n`)
  process.exitCode = 1
})

File: packages/dd-trace/test/llmobs/parity/scenarios/langchain/chat-openai-error.py
from common import finish

from parity_langchain import chat_anthropic, chat_openai, openai_embeddings, start  # noqa: F401


start()
try:
    chat_openai(temperature=0, max_retries=0).invoke([("user", "This request fails.")])
except Exception:
    pass  # expected: the stub returns a 400
finish()
